### PR TITLE
Feature/category UI

### PR DESCRIPTION
**Pull request by Anant-Raman**

### CODE REVIEW

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/CategoryViewPagerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+
+class CategoryViewPagerAdapter(fm: FragmentManager?, lifecycle: Lifecycle) :
+    FragmentStateAdapter(fm!!, lifecycle) {
+
+    private val int_items = 2
+
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/categoryFragment/CategoryViewPagerAdapter.kt:13]: Remove Bang Operator

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/ExpenseCategoryFragment.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.getExpenseCategoryList()
+    }
+
+    private fun setUpObserver() {
+        viewModel.expenseCategoryList.observe(viewLifecycleOwner, OneTimeObserver {
+            list = it.expenseCategoryList
+            initRecyclerView(it.expenseCategoryList)
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/categoryFragment/ExpenseCategoryFragment.kt]: Remove recycler view initialization from observer

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    private val viewModel: IncomeCategoryViewModel by viewModels { viewModelFactory }
+
+    override fun getSnackBarErrorView(): SnackViewErrorBinding = viewBinding.includeIncomeErrorView
+
+    private var list: List<String> = emptyList()
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setUpObserver()
+        setOnCLickListener()
+        viewModel.getIncomeCategoryList()
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt]: Remove from onviewcreated

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.getIncomeCategoryList()
+    }
+
+    private fun setUpObserver() {
+        viewModel.incomeCategoryList.observe(viewLifecycleOwner, OneTimeObserver {
+            list = it.incomeCategoryList
```
Comment (akhilj33) [inline app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt]: same recycler view

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt (hunk 3 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt]: Done

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/CategoryViewPagerAdapter.kt (hunk 1 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/categoryFragment/CategoryViewPagerAdapter.kt:13]: Done

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/ExpenseCategoryFragment.kt (hunk 2 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/categoryFragment/ExpenseCategoryFragment.kt]: Done

File: app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt (hunk 4 above)
Comment (Anant-Raman) [reply app/src/main/java/com/example/mobiledger/presentation/categoryFragment/IncomeCategoryFragment.kt]: Done

**Review by akhilj33 — APPROVED** [review]

**Pull request merged**